FlameBot.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+/**
+ * The most polite bot in the world
+ */
+class FlameBot {
+  /**
+   * Constructs the flame bot
+   * @param {number} flameRate - The chance how often the bot flames back on a message (1 = 100 %)
+   * @param {number} stickerRate - The chance how often the bot replies with a sticker instead of a message (1 = 100 %)
+   * @param {Object} oneLiners - The oneLiners dependency
+   * @param {Object} replies - The replies dependency
+   * @param {Object} telegram - The telegram bot API dependency
+   */
+  constructor(flameRate, stickerRate, oneLiners, replies, telegram) {
+    /**
+     * The chance how often the bot flames back on a message (1 = 100 %)
+     * @type {number}
+     */
+    this.flameRate = flameRate;
+    /**
+     * The chance how often the bot replies with a sticker instead of a message (1 = 100 %)
+     * @type {number}
+     */
+    this.stickerRate = stickerRate;
+    /**
+     * The oneLiners dependency
+     * @type {Object}
+     */
+    this.oneLiners = oneLiners;
+    /**
+     * The replies dependency
+     * @type {Object}
+     */
+    this.replies = replies;
+    /**
+     * The telegram telegram dependency
+     * @type {Object}
+     */
+    this.telegram = telegram;
+  }
+
+  /**
+   * Sets the handlers to start the bot
+   */
+  start() {
+    this.telegram.getMe().then((res) => {
+      /**
+       * The bot’s name
+       * @type {string}
+       */
+      this.name = res.username;
+    });
+
+    this.telegram.on('message', (message) => {
+      this.handleMessage(message);
+    });
+  }
+
+  /**
+   * Replies with an insult
+   *
+   * @param {object} message - The message to reply to
+   * @param {object} user - The user to insult
+   */
+  replyInsult(message, user) {
+    if (Math.random() > this.stickerRate) {
+      const insult = this.oneLiners.getRandomInsult(user.first_name);
+      this.replyText(insult, message);
+    } else {
+      const sticker = this.oneLiners.getRandomSticker();
+      this.telegram.sendSticker(message.chat.id, sticker, { reply_to_message_id: message.message_id });
+    }
+  }
+
+  /**
+   * Replies text to a message
+   *
+   * @param {string} text - The text to send
+   * @param {object} message - The message to reply to
+   */
+  replyText(text, message) {
+    this.telegram.sendMessage(message.chat.id, text, { reply_to_message_id: message.message_id });
+  }
+
+  /**
+   * Handles new messages and replies with insults if necessary
+   *
+   * @param {object} message - The message to reply to
+   */
+  handleMessage(message) {
+    if (this.name === undefined) {
```

Comment:
At least it was. Comparing to the global `undefined` property was discouraged because `undefined` wasn’t reserved and could be set to a different value. Since ECMAScript 5 that’s not possible anymore, so `undefined` is “safe” (barring shadowing through a local variable). The main benefit of still using `typeof` was that it was safe to not throw an Error in any case, but [that changed in ECMAScript 2015](http://es-discourse.com/t/why-typeof-is-no-longer-safe/15) …

So I decided to just use the shorter comparison to `undefined` as in my opinion it’s easier to read. I’m happy to get convinced to change back to `typeof` if somebody knows a reason why. 🙂

(But this line doesn’t matter anymore as 83d1a832f72145a431b003756a7eb1ae124285ea changed the undefined check to a Promise. Not sure if there exists a situation, having Promises and parameter defaults, where a check against undefined is still needed.)